nuxeo-multi-tenant-core/src/test/java/org/nuxeo/ecm/multi/tenant/TestMultiTenantDirectories.java

```diff
@@ -0,0 +1,161 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Gildas Lefevre
+ */
+package org.nuxeo.ecm.multi.tenant;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import javax.inject.Inject;
+import javax.security.auth.login.LoginContext;
+import javax.security.auth.login.LoginException;
+
+import java.security.Principal;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.directory.test.DirectoryFeature;
+import org.nuxeo.ecm.core.api.CloseableCoreSession;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.NuxeoPrincipal;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.ecm.directory.DirectoryException;
+import org.nuxeo.ecm.directory.Session;
+import org.nuxeo.ecm.directory.api.DirectoryService;
+import org.nuxeo.ecm.platform.test.PlatformFeature;
+import org.nuxeo.ecm.platform.usermanager.UserManager;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+/**
+ * Test class for the management of directories with multitenancy.
+ *
+ * @since 10.2
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ DirectoryFeature.class, PlatformFeature.class })
+@Deploy("org.nuxeo.ecm.default.config")
+@Deploy("org.nuxeo.ecm.platform.userworkspace.core")
+@Deploy("org.nuxeo.ecm.multi.tenant")
+@Deploy("org.nuxeo.ecm.multi.tenant:multi-tenant-test-contrib.xml")
+@Deploy("org.nuxeo.ecm.multi.tenant:multi-tenant-enabled-default-test-contrib.xml")
+@Deploy("org.nuxeo.ecm.multi.tenant:multi-tenant-test-directories-contrib.xml")
+@Deploy("org.nuxeo.ecm.platform.test:test-usermanagerimpl/userservice-config.xml")
+@RepositoryConfig(cleanup = Granularity.METHOD, init = MultiTenantRepositoryInit.class)
+public class TestMultiTenantDirectories {
+
+    @Inject
+    protected CoreFeature coreFeature;
+
+    @Inject
+    protected DirectoryService directoryService;
+
+    @Inject
+    protected UserManager userManager;
+
+    @Test
+    public void iCanCreateSameDirectoryEntryInDifferentTenant() throws LoginException {
+        // Add user0 and user1 to the powerusers group so they can edit the directory
+        Framework.doPrivileged(() -> {
+            NuxeoPrincipal user0Principal = userManager.getPrincipal("user0");
+            List<String> groups = user0Principal.getGroups();
+            groups.add(Constants.POWER_USERS_GROUP);
```

Comment:
`POWER_USERS_GROUP` can be import statically (not a need to change it)

Reply:
Done